### DISCUSSION TITLE
How to  redeploy the modified services

### DISCUSSION THREAD

**Discussion by tuhin24** [aws-samples/aws-saas-factory-ref-solution-serverless-saas#65] [Q&A]
I was able to deploy Serverless SaaS solution to my AWS account running from Cloud9. Can you please guide the best way to the modify the lambda services (Order and Product Services), test them locally and deploy the services to my account?

I have tried the following but it was giving error:
 sam deploy  -t tenant-template.yaml --capabilities CAPABILITY_NAMED_IAM     

Error:
Initiating deployment
=====================
File with same data already exists at sam-app/aa4f7c8b33ac987f45aa5d1a153e3058.template, skipping upload
Error: Failed to create changeset for the stack: sam-app, An error occurred (ValidationError) when calling the CreateChangeSet operation: Stack:arn:aws:cloudformation:us-east-2:565524614846:stack/sam-app/5f6fe160-621f-11ee-822c-028a90095031 is in ROLLBACK_COMPLETE state and can not be updated.

**Comment by ujwalbukka** [comment]
Hi @tuhin24 - When you deployed the Serverless SaaS solution an codecommit repo with name "aws-saas-factory-ref-serverless-saas" and a CI/CD pipeline with name "serverless-saas-pipeline" (which has this repo as source) will be created in your AWS account. Check-in your Order and Product Services changes to that repo. And serverless-saas-pipeline will take care of pushing the updates to all tenant stacks.
